src/electionguard/api/filename.c

```diff
@@ -4,6 +4,113 @@
 #define __USE_SECURE_APIS__
 #endif
 
+bool generate_filename(char *path_in, char *prefix_in, char* default_prefix, char *filename_out)
```

Comment:
Looks like a lot of duplicated code, which BTW is not trivial to read due to `#ifdef` (usage of secure API when possible). I would recommend refactoring this code.

Reply:
I was considering opening a ticket to abstract this pattern to make the code more readable in general.  After the last PR's there are other places in the code where we could benefit from using the secure versions of API's on Windows, specifically.  I was considering using macros throughout and handling the compile directives in a single place to improve readability, similar to how you can use `Secure Template Overloads` with the c++ compiler.

What do you think?